haystack/nodes/other/shaper.py

```diff
@@ -0,0 +1,287 @@
+import copy
+import inspect
+from collections import deque
+from typing import Optional, Union, List, Dict, Any, Tuple, Callable
+from types import MappingProxyType
+
+from haystack.nodes.base import BaseComponent
+from haystack.schema import Document, MultiLabel
+
+
+class Shaper(BaseComponent):
+    outgoing_edges = 1
+
+    def __init__(self, inputs: Dict[str, Any]):
+        super().__init__()
+        self.inputs = inputs
+
+        concat: Callable[[List[str]], str] = lambda docs, delimiter=" ": delimiter.join(docs)  # type: ignore
+        concat_docs: Callable[[List[Document]], str] = lambda docs, delimiter=" ", num_tokens=-1: delimiter.join(  # type: ignore
+            [d.content for d in docs]
+        )[
+            :num_tokens
+        ]
+        tmp_registry: Dict[str, Callable] = {
+            "len": len,
+            "expand": (lambda expand_target, size: [expand_target] * size),
+            "expand:size": lambda documents: len(documents),  # pylint: disable=unnecessary-lambda
+            "concat": concat,
+            "concat_docs": concat_docs,
```

Comment:
I think it would be very helpful to have these predefined functions listed in the main docstring of `Shaper` so users would be able to know that they have them as options. And to explain the differences between them. For example, `concat` looks like it is meant to be used with a list of text and `concat_docs` looks like it is meant to be used with a list of Haystack documents.

Reply:
Excellent point @sjrl

Reply:
Done with https://github.com/deepset-ai/haystack/commit/eff2ba4a5493cda17ceb6995b8b8e2355a37e462